util.h

```diff
@@ -43,4 +45,11 @@ bool operator<=(const Type& that) const { return compare3(that) <= 0; } \
 bool operator> (const Type& that) const { return compare3(that) >  0; } \
 bool operator>=(const Type& that) const { return compare3(that) >= 0; }
 
+//  Returns true iff value is representable by (arithmetic) type Target
+template<class Target, class Value>
+constexpr bool representable(Value const& value) {
+	using C = std::common_type_t<Target, Value>;
+	return static_cast<C>(value) <= static_cast<C>(std::numeric_limits<Target>::max());
```

Comment:
There's an implicit assumption here that either `Value` is an unsigned type or `value` is non-negative. Even allowing for that, this isn't testing that `value` is *representable* as a `Target`: it's checking that `value` is within the *representable range* of `Target`.

Change the name to something like `in_representable_range` and verify also that `std::numeric_limits<Target>::min() <= value`.